test/app-luatest/pcall_test.lua

```diff
@@ -0,0 +1,78 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function()
+    g.server = server:new({alias = 'pcall'})
+    g.server:start()
+end)
+
+g.after_all(function()
+    g.server:stop()
+end)
+
+-- #267: Bad exception catching.
+g.test_pcall_inside_xpcall = function()
+    g.server:exec(function()
+        local function pcalltest()
+            local ERRMSG = "module 'some_invalid_module' not found"
+            local status, msg = pcall(require, 'some_invalid_module')
+            if status == false and msg ~= nil and msg:match(ERRMSG) ~= nil then
+                return 'pcall is ok'
+            else
+                return 'pcall is broken'
+            end
+        end
+
+        local status, msg = xpcall(pcalltest, function(msg)
+            print('error handler:', msg)
```

Comment:
Shouldn't we re-raise an error in the case? In the original test reaching the xpcall callback fails the test (because the print statement causes the result file miscompare).

Reply:
Seems so, missed it, thanks!
Conversion of `print` to `error` should be enough, right? If `pcalltest` fail, `xpcall` calls a passed function that will raise.

```diff
--- a/test/app-luatest/pcall_test.lua
+++ b/test/app-luatest/pcall_test.lua
@@ -26,7 +26,7 @@ g.test_pcall_inside_xpcall = function()
         end
 
         local status, msg = xpcall(pcalltest, function(msg)
-            print('error handler:', msg)
+            error('error handler: ' .. msg)
         end)
         t.assert_equals(status, true, 'pcall inside xpcall, status')
         t.assert_equals(msg, 'pcall is ok',
```

Reply:
For me it seems enough on a brief glance.